### DISCUSSION TITLE
Prefer super or crate for internal module references

### DISCUSSION THREAD

**Discussion by huachaohuang** [rust-lang-nursery/api-guidelines#255] [API Guidelines]
We can refer to modules in the same crate in different ways. For example, given a crate like this:

```
src/
  lib.rs
  a.rs
  b.rs
```

`a` can import trait `B` in `b` as `crate::b::B` or `super::b::B`. And if `B` is `pub use` in `lib.rs`, `a` can also refer `B` as `crate::B`. All these work, but it is troublesome to think about which way to go every time. So I think it will be better if we have a guideline about this.

My opinion is that we should prefer referring things relatively using `super::b::B` because:

- It doesn't assume if `B` is public or not
- When refactoring file structure, for example, if we want to move `a.rs` and `b.rs` to a sub-directory, the code still works

**Comment by dtolnay** [comment]
This makes no difference to the API of a crate so I believe it would be off topic for the API Guidelines. This project is not an all-purpose Rust code style guide.

**Reply by huachaohuang** [reply]
Got it. Any recommended place for this discussion?

**Reply by tisonkun** [reply]
@huachaohuang I think it's https://github.com/rust-dev-tools/fmt-rfcs .

Reference
* https://github.com/rust-lang/rust-guidelines
* https://doc.rust-lang.org/1.0.0/style/

**Reply by tisonkun** [reply]
And here it is https://github.com/rust-dev-tools/fmt-rfcs/issues/167.
